### ISSUE TITLE
Unexpected / wrong data type used for storage

### ISSUE THREAD

**Issue by WeilerP** [scverse/anndata#429]
According to the [docs](https://anndata.readthedocs.io/en/latest/anndata.AnnData.html), the keyword `dtype` sets the data type used for storage. However, this seems to only apply to `X`, as the following example shows:

```py
>>> import numpy as np
>>> from anndata import AnnData
>>> 
>>> adata = AnnData(
...     X=np.zeros((2, 2), dtype='int64'),
...     layers={'a': np.zeros((2, 2), dtype='int64')},
...     obsm={'b': np.zeros((2, 2), dtype='int64')},
...     varm={'c': np.zeros((2, 2), dtype='int64')},
... )
>>> adata.X.dtype
dtype('float32')
>>> adata.layers["a"].dtype
dtype('int64')
>>> adata.obsm["b"].dtype
dtype('int64')
>>> adata.varm["c"].dtype
dtype('int64')
```

Is this really the intended behaviour or a bug?

**Comment by ivirshup** [comment]
Pretty much. We'd like to remove the argument and just use whatever dtype `X` already is though.

**Comment by WeilerP** [comment]
Okay, thanks! So, until the new feature has been implemented, it might be worth stating it more clearly in the docs to rule out any confusion / false expectations?

**Comment by flying-sheep** [comment]
Fixed in #551 and #854
